p2p/dandelion-tower/src/pool/incoming_tx.rs

```diff
@@ -0,0 +1,112 @@
+//! Contains [`IncomingTx`] and [`IncomingTxBuilder`]
+use crate::{State, TxState};
+
+/// An incoming transaction that has gone through the preprocessing stage.
+pub struct IncomingTx<Tx, TxId, PId> {
+    /// The transaction.
+    pub(crate) tx: Tx,
+    /// The transaction ID.
+    pub(crate) tx_id: TxId,
+    /// The routing state of the transaction.
+    pub(crate) routing_state: TxState<PId>,
+}
+
+/// An [`IncomingTx`] builder.
+///
+// The const generics here are used to restrict what methods can be called.
+// RS: a bool for if the routing state is set.
+// DBS: a bool for if the state in DB is set.
+pub struct IncomingTxBuilder<const RS: bool, const DBS: bool, Tx, TxId, PId> {
```

Comment:
> these are just an implementation detail so probably don't need to be in public docs

Isn't it part of the public API? It's not a big deal here but I don't think it hurts to add a couple doc lines to make it immediately clear what these are.

```suggestion
/// An [`IncomingTx`] builder.
///
/// The const generics here are used to restrict what methods can be called.
///
/// - `RS`: routing state; a `bool` for if the routing state is set
/// - `DBS`: database state; a `bool` for if the state in the DB is set
pub struct IncomingTxBuilder<const RS: bool, const DBS: bool, Tx, TxId, PId> {
```